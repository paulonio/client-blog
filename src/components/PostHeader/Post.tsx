import React, { FC } from 'react';
import { PostType } from '@/types/types';

import styles from './styled.module.scss';
import { sen } from '@/styles/fonts';

interface PostProps {
  post: PostType;
}

const Post: FC<PostProps> = ({ post }) => {
  const { username, date, title, body, urlToImage } = post;

  return (
    <div className={styles.post__wrapper}>
      <div className={styles.post__image}>
        <img src={urlToImage} alt={`${username} post`} />
      </div>
      <div className={styles.post__content}>
        <p className="body-l">
          By <span className={styles.username}>{username}</span> | {date}
        </p>
        <h3 className={sen.className}>{title}</h3>
        <p className="body-s">{body}</p>
      </div>
    </div>
  );
};

export default Post;
