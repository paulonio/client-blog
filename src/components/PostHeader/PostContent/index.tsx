import React from 'react';
import { useTranslation } from 'next-i18next';

import { sen } from '@/styles/fonts';

import styles from '../styled.module.scss';

const PostContent = () => {
  const { t } = useTranslation('blog');

  return (
    <section className={styles['single-post']}>
      <h2 className={sen.className}>{t('lorem-ipsum')}</h2>
      <p className="body-s">{t('body')}</p>
      <h2 className={sen.className}>{t('step-by-step-guide')}</h2>
      <p className="body-s">{t('body')}</p>
      <p className="body-s">{t('body')}</p>
      <ul>
        <li>
          <h4 className={sen.className}>{t('list.item-1')}</h4>
        </li>
        <li>
          <h4 className={sen.className}>{t('list.item-2')}</h4>
        </li>
        <li>
          <h4 className={sen.className}>{t('list.item-3')}</h4>
        </li>
      </ul>
      <p className="body-s">{t('body')}</p>
      <h2 className={sen.className}>{t('lorem-ipsum')}</h2>
      <p className="body-s">{t('body')}</p>
    </section>
  );
};

export default PostContent;
