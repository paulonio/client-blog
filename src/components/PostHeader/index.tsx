import React, { FC } from 'react';
import { useTranslation } from 'next-i18next';

import styles from './styled.module.scss';
import { PostType } from '@/types/types';
import { sen } from '@/styles/fonts';
import { stringToKey } from '@/utils/utils';

interface PostHeaderProps {
  post: PostType;
}

const PostHeader: FC<PostHeaderProps> = ({ post }) => {
  const { t } = useTranslation(['common', 'blog']);
  const { urlToAvatar, username, date, category, title } = post;

  return (
    <section className={styles.post__title}>
      <div className={styles.profile}>
        <div className={styles.avatar}>
          <img src={urlToAvatar} alt={username} />
        </div>
        <div>
          <h4 className={styles.username}>{t(stringToKey(username))}</h4>
          <p className="body-s">
            {t('posted-on', { ns: 'blog' })} {t(stringToKey(date))}
          </p>
        </div>
      </div>
      <h1 className={styles.title}>{t(stringToKey(title), { ns: 'blog' })}</h1>
      <h4 className={sen.className}>{t(stringToKey(category))}</h4>
    </section>
  );
};

export default PostHeader;
