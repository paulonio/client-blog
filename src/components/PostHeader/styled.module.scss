@import '../../styles/sass/constants';

.post{
  &__wrapper {
    width: 32%;
    height: 100%;
    max-width: 405px;
    transition: background-color 0.3s;

    @media screen and (max-width: 768px) {
      width: 100%;
      max-width: 768px;
      padding: 10px;
    }

    &:hover {
      background-color: $color-light-s;
      cursor: pointer;
    }
  }

  &__title {
    width: 100%;
    max-width: 768px;
    margin: 128px auto 64px;
    display: flex;
    flex-direction: column;
    align-items: start;
    row-gap: 24px;

    @media screen and (max-width: 768px) {
      margin: 32px auto 32px;
    }
  }

  &__content {
    margin-top: 32px;
    display: flex;
    flex-direction: column;
    align-items: start;
    row-gap: 16px;
  }

  &__image {
    max-height: 318px;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;
  }
}

.title {
  margin-bottom: 10px;
  color: $color-black;
}

.profile {
  display: flex;
  align-items: center;
  justify-content: start;
  gap: 16px;
}

.avatar {
  width: 48px;
  height: 48px;
  border-radius: 50px;
  overflow: hidden;

  & img {
    width: 100%;
    height: 100%;
  }
}

.username {
  color: $color-secondary;
}

.image__wrapper {
  width: 100%;
  height: auto;

  & img {
    width: 100%;
    height: 100%;
  }
}

.single-post {
  width: 100%;
  max-width: 843px;
  margin: 0 auto;

  & h2 {
    margin: 48px 0 16px;
  }

  & ul {
    margin: 24px 0;
  }
}

.other {
  margin-top: 128px;

  @media screen and (max-width: 768px) {
    margin-top: 32px;
  }

  & h2 {
    color: #000000;
  }

  &__wrapper {
    padding: 48px 0;
    display: flex;
    align-items: stretch;
    gap: 32px;
    border-bottom: solid 1px rgba($color-dark-s, 0.4);

    @media screen and (max-width: 768px) {
      flex-direction: column;
      text-align: center;
    }
  }
}