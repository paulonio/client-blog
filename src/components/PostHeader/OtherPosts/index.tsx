import React, { FC } from 'react';
import { useTranslation } from 'next-i18next';

import Post from './Post';

import type { PostType } from '@/types/types';
import { sen } from '@/styles/fonts';

import styles from './styled.module.scss';

interface OtherPostsProps {
  posts: PostType[];
}

const OtherPosts: FC<OtherPostsProps> = ({ posts }) => {
  const { t } = useTranslation('blog');

  return (
    <section className={styles.other}>
      <h2 className={sen.className}>{t('next-posts')}</h2>
      <div className={styles.other__wrapper}>
        {posts.map((post) => (
          <Post key={post.id} post={post} />
        ))}
      </div>
    </section>
  );
};

export default OtherPosts;
