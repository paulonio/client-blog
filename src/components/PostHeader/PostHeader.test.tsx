import React from 'react';
import { render } from '@testing-library/react';
import PostHeader from '.';
import type { PostType } from '@/types/types';

const TEST_DATA: PostType = {
  id: 'test',
  body: 'test',
  category: 'test',
  date: 'test',
  tags: ['test'],
  title: 'test',
  urlToAvatar: 'test',
  urlToImage: 'test',
  username: 'test',
};

describe('Post header', () => {
  it('should render post header', () => {
    const { container } = render(<PostHeader post={TEST_DATA} />);

    expect(container).toMatchSnapshot();
  });
});
