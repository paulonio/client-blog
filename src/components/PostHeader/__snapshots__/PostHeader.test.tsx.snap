// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post header should render post header 1`] = `
<div>
  <section
    class="post__title"
  >
    <div
      class="profile"
    >
      <div
        class="avatar"
      >
        <img
          alt="test"
          src="test"
        />
      </div>
      <div>
        <h4
          class="username"
        >
          test
        </h4>
        <p
          class="body-s"
        >
          posted-on
           
          test
        </p>
      </div>
    </div>
    <h1
      class="title"
    >
      test
    </h1>
    <h4
      class="className"
    >
      test
    </h4>
  </section>
</div>
`;
